Add PostInitializationEvent

DrkMatr1984 opened the pull request:
This allows one to get the custom item settings immediately after
they're imported from Addons. I needed it for ShinyItems integration for
the Glowstone Sword in RPGAdditions.

https://github.com/MinevoltRPG/RPGAdditions/blob/f1490ad92c6b4ca85b4b147971afb330982cd581/src/main/java/me/drkmatr1984/RPGAdditions/ShinyItemsHook.java#L23

chrismin13 commented:
No need for another listener, just change the @EventHandler to @EventHandler(priority = EventPriority.HIGHEST) so that my plugin had finished adding all the Custom Items. You can even modify then without causing issues if you need to!

chrismin13 commented:
Good point though for making sure the Durabilities are set correctly. You can do a scheduleSyncDelayedTask, but this seems like a more elegant solution.

DrkMatr1984 commented:
Yup, I'm cleaning some code up now and getting rid of that extra ugly listener. I usually just slop some code together for functionality, and then clean up later :P This worked for the moment, but I've already cleaned up on my local copy and dropped it down to one listener. I thought about a scheduled task also, but that adds extra burden to the CPU when I could just add a PostInit listener to AdditionsAPI that triggers once I'm 100% sure all the items are imported and added. Plus, depending on the plugins on the server and the speed of the server itself, I was unsure when to set a scheduled task to execute as the load time may differ on different server setups. A PostInit event seemed much more reliable.
